### DISCUSSION TITLE
Converting snake_case to CamelCase: what will be the best approach?

### DISCUSSION THREAD

**Discussion by grreeenn** [kubb-labs/kubb#1430] [Q&A] [answered]
Hi :)

I'm currently working on replacing the view layer of a RoR project to React SPA.
Kubb is doing a great job of adding a robust layer between the API and the client; however, Ruby and JS have different naming conventions — Ruby uses snake_case, while JS uses camelCase.

The team is furious seeing snake_case in JS code 😆 we found a way to convert the case in runtime with Axios middleware, now I need my types to match the actual axios output.

What will be the best way to approach this with Kubb? Should I make a PR for a new option in config, or it'd be better to create my own plugin based on plugin-ts? Or can generators help me here (I only saw the examples of adding new files with them, not hooking into the generation)?

**Comment by stijnvanhulle** [comment, marked as answer]
@grreeenn hi, maybe this library could help in already formatting to format that Kubb understands: https://github.com/thim81/openapi-format. If the change of casing does not work, we could have a look in changing Kubb.

**Reply by grreeenn** [reply]
great tool, and looks like it does exactly what I want! Thank you :)))
